optaplanner-core/src/main/java/org/optaplanner/core/impl/domain/entity/descriptor/EntityDescriptor.java

```diff
@@ -543,7 +567,16 @@ public long getMaximumValueCount(Solution_ solution, Object entity) {
     public long getProblemScale(Solution_ solution, Object entity) {
         long problemScale = 1L;
         for (GenuineVariableDescriptor<Solution_> variableDescriptor : effectiveGenuineVariableDescriptorList) {
-            problemScale *= variableDescriptor.getValueCount(solution, entity);
+            long valueCount = variableDescriptor.getValueCount(solution, entity);
+            problemScale *= valueCount;
+            if (variableDescriptor.isListVariable()) {
+                // This formula probably makes no sense other than that it results in the same problem scale for both
+                // chained and list variable models.
+                // TODO fix https://issues.redhat.com/browse/PLANNER-2623 to get rid of this.
+                problemScale *= valueCount;
+                problemScale /= getSolutionDescriptor().getEntityCount(solution);
+                problemScale += valueCount;
+            }
         }
         return problemScale;
```

Comment:
Is this unit-tested? It's just obscure enough that it should be. :-)

Reply:
Yes, it is. Although this is temporary, I wouldn't dare publishing something like this without a proof that it does what it's expected to :-)